### PR TITLE
vispy: make VispyWidget a first class object

### PR DESCRIPTION
**Pull request by cgestes**
is that the right direction?

### CODE REVIEW

File: vispy/ipython/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014, 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+from .widget import VispyWidget
```
Comment (rossant) [inline vispy/ipython/__init__.py:5]: we should probably move `vispy/ipython/` to `vispy/app/backends/_ipython/`

File: vispy/app/backends/_ipynb_webgl.py [hunk 2]
```diff
@@ -137,11 +136,15 @@ def __init__(self, *args, **kwargs):
 
         self._create_widget(size=p.size)
 
+    def set_widget(self, widget):
+        self._widget = widget;
+        self._vispy_canvas.context.shared.parser._widget = widget;
```
Comment (rossant) [inline vispy/app/backends/_ipynb_webgl.py:141]: why is this line needed?

File: vispy/ipython/widget.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014, 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+
+from IPython.html.widgets import DOMWidget
+from IPython.utils.traitlets import Unicode, Int, Bool
+from vispy.app.backends._ipynb_util import create_glir_message
```
Comment (rossant) [inline vispy/ipython/widget.py:7]: move `_ipynb_util` to `vispy/app/backends/_ipython/`

Comment (cgestes) [inline vispy/ipython/widget.py:7]: dont you want
from vispy.ipython import VispyWidget

in your notebook app?

maybe we can create the canvas directly from the widget or even make the
widget derive from the widget ideas?

Comment (rossant) [inline vispy/ipython/widget.py:7]: > from vispy.ipython import VispyWidget

hmm I'd prefer `from vispy.app.backends.ipython import VispyWidget`

> maybe we can create the canvas directly from the widget or even make the
> widget derive from the widget ideas?

I don't understand can you explain?

Comment (cgestes) [inline vispy/ipython/widget.py:7]: Ok for app.backends.ipython.

humm nevermind my idea lead to code that only work with a specific backend of vispy...

will try to fix size now. seems more complicated than it appear!
